src/item.h

```diff
@@ -670,6 +677,45 @@ class item : public JsonSerializer, public JsonDeserializer, public visitable<it
      */
     int chip_resistance( bool worst = false ) const;
 
+    /** How much damage has the item sustained? */
+    int damage() const {
+        return damage_;
+    }
+
+    /** Minimum amount of damage to an item (state of maximum repair) */
+    int min_damage() const;
+
+    /** Maximum amount of damage to an item (state before destroyed) */
+    int max_damage() const;
+
+    /**
+     * Apply damage to item constrained by @ref min_damage and @ref max_damage
+     * @param qty maximum amount by which to adjust damage (negative permissible)
+     * @param dmg type of damage which may be passed to @ref on_damage callback
+     * @return whether item should be destroyed
+     */
+    bool mod_damage( int qty, damage_type dt = DT_NULL );
```

Comment:
> would require somehow storing the destroyed status within the item

It won't. All the things are already here: 
```c++
bool item::destroyed() const { 
    return damage_ > max_damage(); // Just cap damage in the getter
}
```
> This version is reasonably fail-safe in that failing to check the return results in an effective no-op with the resulting item guaranteed to have damage(). In the above more verbose suggestion the same failure to correctly handle the return results in a partly-destroyed intermediate state that can (and probably will) be copied around.

I see no differences here. You get no-op in any case, but don't copy around a _completely_ destroyed item: instead it can (should) be removed it in one known place while processing the turn.